### PR TITLE
Fixing Thief Beacon Role Check Logic (to use new mindrole system)

### PR DESCRIPTION
**Pull request by sporkyz**
## About the PR

The Thief Beacon currently directly checks if the user has the thief role instead of using the new mindrole system. This means that on master, the thief beacon currently doesn't give users the dialogue option to set its coordinates, even if they're a thief. I've updated the thief beacon logic to use the new mindrole system, fixing this bug. 
## Why / Balance

Thieves should be able to set their beacon coordinates as intended. Fixes https://github.com/space-wizards/space-station-14/issues/32763
## Technical details

Added a using declaration for the content shared roles system to allow the use of MindHasRole. Replaced the existing HasComp check with MindHasRole.
## Media

image
(just trust me the set coordinates option was missing before my changes)

## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

### CODE REVIEW

File: Content.Server/Thief/Systems/ThiefBeaconSystem.cs
Comment (Errant-4) [inline Content.Server/Thief/Systems/ThiefBeaconSystem.cs:45]: You can combine the two if statements with an OR operator

if (mind == null || !_roles.MindHasRole<ThiefRoleComponent>(mind.Value))
            return;

**Review by Errant-4 — CHANGES_REQUESTED** [review]
Hey, thanks a lot for finding this issue! I just have a minor formatting request and then it's good to go

**Comment by sporkyz** [comment]
I've added the requested formatting change. Thanks for the feedback!

**Review by Errant-4 — APPROVED** [review]

**Review by slarticodefast — APPROVED** [review]
Tested in-game and works as intended.
This should be merged into stable/staging as a critical bugfix.
So we need a third maintainer approval.

**Review by Jezithyr — APPROVED** [review]

**Comment by slarticodefast** [comment]
Uhh, changing the base branch to staging added more changes for some reason. DON'T MERGE until I figure this out

**Comment by slarticodefast** [comment]
Ok, I reverted the base back to master and the unwanted changes are gone. According to Julian we need to cherry pick this into staging. If any maintainer could do this that would be great. I'm gonna learn more git in the meantime 😅

**Pull request merged**
